src/client/datascience/notebook/notebookEditor.ts

```diff
@@ -311,17 +313,14 @@ export class NotebookEditor implements INotebookEditor {
     }
 
     private runCellRange(cells: NotebookCell[]) {
-        const kernel = this.kernelProvider.get(this.file);
-
-        if (!kernel || this.restartingKernel) {
+        if (cells.length === 0) {
             return;
         }
-
-        cells.forEach(async (cell) => {
-            if (cell.kind === NotebookCellKind.Code) {
-                await kernel.executeCell(cell);
-            }
-        });
+        const controller = this.controllerManager.getSelectedNotebookController(cells[0].notebook);
+        if (!controller || this.restartingKernel) {
+            return;
+        }
+        controller.handleExecution(cells.filter((cell) => cell.kind === NotebookCellKind.Code)).catch(noop);
```

Comment:
Should the noop be a traceError or traceInfo here? Just so we have something in the log if this fails unexpectedly.

Reply:
Done